structuralcodes/materials/reinforcement/_reinforcement.py

```diff
@@ -85,20 +88,59 @@ def epsuk(self, epsuk: float) -> None:
         """Setter for epsuk."""
         self._epsuk = abs(epsuk)
 
+    @property
+    def epsyk(self) -> float:
+        """Returns characteristic yield strain epsyk."""
+        return self.fyk / self.Es
+
     @property
     def constitutive_law(self) -> ConstitutiveLaw:
         """Returns the constitutive law object."""
         return self._constitutive_law
 
     @constitutive_law.setter
-    def constitutive_law(self, constitutive_law: ConstitutiveLaw) -> None:
-        """Setter for constitutive law."""
-        if 'rebars' in constitutive_law.__materials__:
-            self._constitutive_law = constitutive_law
+    def constitutive_law(
+        self,
+        constitutive_law: t.Union[
+            ConstitutiveLaw,
+            t.Literal[
+                'elastic',
+                'elasticperfecltyplastic',
+                'elasticplastic',
+            ],
+        ],
+    ) -> None:
+        """Setter for constitutive law.
+
+        Args:
+        consitutive_law (ConstitutiveLaw): a valid ConstitutiveLaw object
+                for reinforcement (optional)
+        constitutive_law_name (str): a string defining a valid constitutive law
+                type for reinforcement.
+                (valid options: 'elastic', 'elasticperfectlyplastic',
+                'elasticplastic').
```

Comment:
Remember to update the docstring to match the signature.

Reply:
Done in 16967e2fe4a6290daab9551cb04ae7167cc69c21